nixos/tests/nginx-media-types.nix

```diff
@@ -0,0 +1,26 @@
+import ./make-test-python.nix ({lib, pkgs, ...}:
+{
+  name = "nginx-media-types";
+  meta.maintainers = with pkgs.lib.maintainers; [ izorkin ];
+
+  nodes = {
+    machine = { pkgs, ... }: {
+      services.nginx = {
+        enable = true;
+        defaultMimeTypes = "${pkgs.media-types}/etc/nginx/mime.types";
+      };
+    };
+  };
+
+  testScript = ''
+    machine.start()
+
+    machine.wait_for_unit("nginx")
+    machine.wait_for_open_port(80)
+
+    # Checking for duplicate media.types
+    machine.fail("journalctl --unit nginx --grep 'duplicate extension'")
```

Comment:
Not so super happy with this, we are essentially relying on a warning which wouldn't be triggered in an ideal world and if this test would fail, the software could actually be in a better state

Reply:
Can only be checked by running the nginx server. I don't know any other way.